src/vendor/inc/c_link.pxd

```diff
@@ -52,6 +52,10 @@ cdef extern from "azure_uamqp_c/link.h":
     int link_get_max_message_size(LINK_HANDLE link, stdint.uint64_t* max_message_size)
     int link_get_peer_max_message_size(LINK_HANDLE link, stdint.uint64_t* peer_max_message_size)
     int link_set_attach_properties(LINK_HANDLE link, c_amqp_definitions.fields attach_properties)
+    int link_set_desired_capabilities(LINK_HANDLE link, c_amqpvalue.AMQP_VALUE desired_capabilities)
+    int link_get_desired_capabilities(LINK_HANDLE link, c_amqpvalue.AMQP_VALUE* desired_capabilities)
+    int link_set_offered_capabilities(LINK_HANDLE link, c_amqpvalue.AMQP_VALUE offered_capabilities)
```

Comment:
I don't think these will ever be set, only retrieved. So maybe only have the one function:
`link_get_peer_offered_capabilities`

Reply:
What I am thinking is, if someone want to use our uamqp-python library to establish uamqp server and client, then we should expose both setting/getting on these two properties.

My understanding is that `offered_capabilities` is for the server role to set, and the `desired_capabilities` is for the client role to set.